lib/rubocop/cop/style/any_predicate.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Prefer OR expressions over `Array#any?`.
+      #
+      # This cop only checks when the receiver of `any?` is an array literal.
+      #
+      # @safety
+      #   Autocorrection is unsafe because `any?` returns a boolean while the
+      #   OR expression returns the first truthy value.
+      #
+      #   It's also unsafe because OR expressions short-circuit; if any element
+      #   is a method call with side effects, they may not run anymore with an
+      #   OR expression.
+      #
+      #   [source,ruby]
+      #   ----
+      #   [foo, method_with_side_effects].any?
+      #   # always evaluates `method_with_side_effects`
+      #
+      #   foo || method_with_side_effects
+      #   # evaluates `method_with_side_effects` only if `foo` is falsey
+      #   ----
+      #
+      # @example
+      #
+      #   # bad
+      #   [foo, bar, baz].any?
+      #
+      #   # good
+      #   (foo || bar || baz)
+      #
+      #   # good - arrays with splat arguments are ignored
+      #   [foo, *bar].any?
+      #
+      class AnyPredicate < Base
```

Comment:
I'm not a fan of the name myself either. Perhaps `OrLikeArrayAny`? I'm open to ideas.

Reply:
ArrayLiteralAny perhaps?

Reply:
Updated to `ArrayLiteralAsLogicalExpression` and added support for `Array#and?`. Perhaps a bit verbose, but let me know what you think.